tech-debt: tests/test_migrate_users.py fails with ModuleNotFoundError: neo4j (CI baseline red)

## Problem

`tests/test_migrate_users.py` imports `scripts.migrate_users`, which imports `neo4j`. The `neo4j` package is not in `pyproject.toml`'s default dependencies, so CI's `uv run pytest` fails at collection time:

```
tests/test_migrate_users.py:9: in <module>
    from scripts.migrate_users import (
scripts/migrate_users.py:22: in <module>
    from neo4j import GraphDatabase
E   ModuleNotFoundError: No module named 'neo4j'
```

This has been red on `main` for the last 3 CI runs (verified via `gh run list --branch main`). Was called out by name in noorinalabs/noorinalabs-main#109's body as a W7 carry-forward.

## Fix options

1. **Add `neo4j` to dev dependencies** — `uv add --dev neo4j` so test collection works.
2. **Move `migrate_users.py` to a separate optional extra** — `uv add --optional migrate neo4j` and skip the test if `neo4j` isn't installed (`pytest.importorskip("neo4j")`).
3. **Conditional import** — guard the `from neo4j import GraphDatabase` with try/except and skip the test when unavailable.

Option 1 is the cheapest and most honest: if the test exists in the repo, its dependencies should be installable in CI. Option 2/3 adds complexity for a script that almost certainly always needs neo4j to function.

## Scope

Belongs to noorinalabs/noorinalabs-main#111 (W8 CI sweep). Does not block #110/PR #58.

## Refs

- Discovered (re-confirmed) during PR #58 review for noorinalabs/noorinalabs-main#110
- Cross-ref: noorinalabs/noorinalabs-main#109

--- a/.github/workflows/ghcr-publish.yml
+++ b/.github/workflows/ghcr-publish.yml
@@ -61,6 +61,7 @@ jobs:
             type=raw,value=latest,enable=${{ github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/v') }}
 
       - name: Build and push
+        id: build
         uses: docker/build-push-action@14487ce63c7a62a4a324b0bfb37086795e31c6c1 # v6.16.0
         with:
           context: .
@@ -74,7 +75,12 @@ jobs:
       - name: Run Trivy vulnerability scan
         uses: aquasecurity/trivy-action@57a97c7e7821a5776cebc9bb87c984fa69cba8f1 # v0.35.0
         with:
-          image-ref: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${{ github.ref_name }}
+          # Scan by digest — guaranteed to exist immediately after `Build and push`
+          # and not subject to tag-resolution races. Previously used
+          # `:${{ github.ref_name }}` which on a `main` push resolves to `:main`,
+          # but the metadata-action emits only `:latest` and `:sha-XXXX` for
+          # main pushes — so the scan always 404'd with MANIFEST_UNKNOWN.
+          image-ref: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}@${{ steps.build.outputs.digest }}
           format: table
           exit-code: "1"
           severity: CRITICAL,HIGH

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -39,6 +39,7 @@ dev = [
     "types-python-jose>=3.3.0",
     "types-pyasn1>=0.6.0",
     "aiosqlite>=0.20.0",
+    "neo4j>=6.1.0",
 ]
 
 [tool.ruff]

